v2/internal/reconcilers/generic/generic_reconciler.go

```diff
@@ -351,12 +401,39 @@ func (gr *GenericReconciler) CommitUpdate(
 }
 
 func (gr *GenericReconciler) handleSkipReconcile(ctx context.Context, log logr.Logger, obj genruntime.MetaObject) error {
-	reconcilePolicy := reconcilers.GetReconcilePolicy(obj, log, gr.Config.DefaultReconcilePolicy) // TODO: Pull this whole method up here
+
+	var reconcilePolicy annotations.ReconcilePolicyValue
```

Comment:
It feels like this whole section where we read the reconcilePolicy is somewhat duplicated - can we factor into a shared method on `gr` that does the hierarchical lookup, and then reuse that helper function in all the locations?

Reply:
Actually I think that's what `reconcilers.GetReconcilePolicy` was trying to do. Maybe delete that and make a method on `gr` that has the hierarchical lookup you want?

Reply:
I think if you do that, `ParseReconcilePolicy` (that honors the default) isn't needed anymore, because that logic is encoded in the `gr` helper instead.